modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -502,54 +465,44 @@ public TableManager(
 
-        configuredTablesCache = new ConfiguredTablesCache(tablesCfg, getMetadataLocallyOnly);
-
         raftCommandsMarshaller = new ThreadLocalPartitionCommandsMarshaller(clusterService.serializationRegistry());
 
         startVv = new IncrementalVersionedValue<>(registry);
     }
 
     @Override
     public void start() {
-        mvGc.start();
+        inBusyLock(busyLock, () -> {
+            mvGc.start();
 
-        lowWatermark.start();
+            lowWatermark.start();
 
-        fireCreateTablesOnManagerStart();
+            startTables();
 
-        metaStorageMgr.registerPrefixWatch(ByteArray.fromString(PENDING_ASSIGNMENTS_PREFIX), pendingAssignmentsRebalanceListener);
-        metaStorageMgr.registerPrefixWatch(ByteArray.fromString(STABLE_ASSIGNMENTS_PREFIX), stableAssignmentsRebalanceListener);
-        metaStorageMgr.registerPrefixWatch(ByteArray.fromString(ASSIGNMENTS_SWITCH_REDUCE_PREFIX), assignmentsSwitchRebalanceListener);
+            metaStorageMgr.registerPrefixWatch(ByteArray.fromString(PENDING_ASSIGNMENTS_PREFIX), pendingAssignmentsRebalanceListener);
+            metaStorageMgr.registerPrefixWatch(ByteArray.fromString(STABLE_ASSIGNMENTS_PREFIX), stableAssignmentsRebalanceListener);
+            metaStorageMgr.registerPrefixWatch(ByteArray.fromString(ASSIGNMENTS_SWITCH_REDUCE_PREFIX), assignmentsSwitchRebalanceListener);
 
-        tablesCfg.tables().listenElements(new ConfigurationNamedListListener<>() {
-            @Override
-            public CompletableFuture<?> onCreate(ConfigurationNotificationEvent<TableView> ctx) {
-                return onTableCreate(ctx);
-            }
+            catalogService.listen(CatalogEvent.TABLE_CREATE, (parameters, exception) -> {
+                assert exception == null : parameters;
 
-            @Override
-            public CompletableFuture<?> onRename(ConfigurationNotificationEvent<TableView> ctx) {
-                // TODO: IGNITE-15485 Support table rename operation.
+                return onTableCreate((CreateTableEventParameters) parameters).thenApply(unused -> false);
+            });
 
-                return completedFuture(null);
-            }
+            catalogService.listen(CatalogEvent.TABLE_DROP, (parameters, exception) -> {
+                assert exception == null : parameters;
 
-            @Override
-            public CompletableFuture<?> onDelete(ConfigurationNotificationEvent<TableView> ctx) {
-                return onTableDelete(ctx);
-            }
-        });
+                return onTableDelete(((DropTableEventParameters) parameters)).thenApply(unused -> false);
+            });
 
-        schemaManager.listen(SchemaEvent.CREATE, new EventListener<>() {
-            @Override
-            public CompletableFuture<Boolean> notify(SchemaEventParameters parameters, @Nullable Throwable exception) {
+            schemaManager.listen(SchemaEvent.CREATE, (parameters, exception) -> inBusyLockAsync(busyLock, () -> {
```

Comment:
Seems, no one listens for SchemaEvent and TableEvent.

Reply:
I'll get rid of them well.

Reply:
I looked at the opportunity to get rid of this, I realized that I need to make quite a few changes for this, I propose to do it in IGNITE-20339.